### PR TITLE
[528] change the layout of the user page on the support console

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

- [This trello ticket](https://trello.com/c/ZRWIM8fK/528-change-the-layout-of-the-user-page-on-the-support-console)

### Changes proposed in this pull request

- Introduce a navigation bar within the user show page
- Add user details as a Summary List Component on the show page and split the providers into a different tab
- Add an edit user details page to edit first and lastname, email and admin status as a checkbox

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by stellenberger** [comment]
[Change the layout of the user page on the Support Console](https://trello.com/c/ZRWIM8fK/528-change-the-layout-of-the-user-page-on-the-support-console)

File: config/routes.rb [hunk 1]
```diff
@@ -57,7 +57,12 @@
       resources :courses, only: %i[index edit update]
       resources :locations
     end
-    resources :users, only: %i[index show new create destroy]
+
+    resources :users do
+      scope module: :users do
+        get :providers, on: :member, to: "providers#show"
```
Comment (darokel) [inline config/routes.rb:63]:
```suggestion
resources :providers, only: :show
```

then you don't have to map?

File: spec/features/support/users/editing_a_user.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+feature "Editing a user" do
+  before do
+    given_i_am_authenticated(user: admin)
+    and_a_user_exists
+    when_i_visit_the_user_edit_page
+  end
+
+  scenario "successfully changing details" do
+    when_i_fill_in_correct_details
```
Comment (darokel) [inline spec/features/support/users/editing_a_user.rb:13]: i'd reword this to begin with `and` so it flows with the steps in the before block

File: spec/features/support/users/editing_a_user.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    and_click_update
+    i_am_taken_to_the_user_show_page
+    with_a_success_message
+  end
+
+  scenario "incorrect details" do
+    when_i_fill_in_with_blank_details
```
Comment (darokel) [inline spec/features/support/users/editing_a_user.rb:20]: same for this one

**Review by darokel — APPROVED** [review]
Code is waaay cleaner now good job 👍 . I've left a couple of comments but have approved this

File: config/routes.rb (hunk 1 above)
Comment (stellenberger) [reply config/routes.rb:63]: Good spot, commited 👍

**Review by mbocc — APPROVED** [review]
Looks great, thank you for such detailed work!
